projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -32,179 +39,243 @@
-
-  // User
-  private String userPrivateName = "שם פרטי";
-  private String userLastName = "שם משפחה";
-  private String userID = "ת ז משתמש";
-  private String userAddress = "כתובת משתמש";
-  private String userPhone = "טלפון משתמש";
-  private String userFax = "פקס משתמש";
-
-  // First spam company's details
-  private String companyName = "שם החברה";
-  private String companyId = "מספר ח.פ";
-  private String companyAddress = "כתובת החברה";
-  private String companyPhone = "טלפון החברה";
-  private String companyFax = "פקס חברה";
-
-  // Second spam company's details
-  private String company2Name = "שם החברה 2";
-  private String company2Id = "מספר ח.פ 2";
-  private String company2Address = "כתובת החברה 2";
-  private String company2Phone = "טלפון החברה 2";
-  private String company2Fax = "פקס חברה 2";
-  // endregion
+      "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש  \nלקבלת דבר/י הפרסומת";
 
+  EditText receivedDate;
+  DatePickerDialog datePickerDialog;
+  Calendar calendar;
   Button createPdfButton;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_lawsuit_pdf);
-    createPdfButton = (Button) findViewById(R.id.button_createPDF);
 
-    initDirStructure();
+    createPdfButton = findViewById(R.id.button_createPDF);
+    receivedDate = findViewById(R.id.editText_spamDate);
+
     setTimeZones();
+    createPdfButton.setOnClickListener(
+        v -> {
+          checkPermissions();
+          if (permissionGranted) {
+            sharePdf(createPdf());
+          }
+        });
+    receivedDate.setOnClickListener(v -> displayDatePicker());
+  }
 
-    createPdfButton.setOnClickListener(v -> checkPermissionsThenCreatePdf());
+  // Used for date fields in the lawsuit
+  private void setTimeZones() {
+    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
   }
 
-  private void createPdf() {
-    // TODO: Success reading license file only from "res/raw". Perhaps should be read from different
-    // dir
-    // Activate iText license
-    LicenseKey.loadLicenseFile(getResources().openRawResource(R.raw.itextkey));
+  // Save PDF to device. Return absFilename of the file
+  private String createPdf() {
+
+    // Read and fill template
+    String template = null;
     try {
-      HtmlConverter.convertToPdf(
-          fillTemplate(),
-          new FileOutputStream(Paths.get(lawsuitOutputPath, getPdfFilename()).toString()));
-      Toast.makeText(LawsuitPdfActivity.this, "Lawsuit created!", Toast.LENGTH_LONG).show();
+      template = fillTemplate();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
 
-      // Read template / write output
+    PdfDocument document = new PdfDocument();
+
+    // Create 2 pages for the PDF (Size A4)
+    PdfDocument.PageInfo firstPageInfo = new PdfDocument.PageInfo.Builder(595, 842, 1).create();
```

Comment:
Import 'PdfDocument.PageInfo' so you don't need to write it every line. Same for 'PdfDocument.page'.

Reply:
Done. :+1: 
Thanks